docs/tutorials/using_apis/eventqueue-background.md

```diff
@@ -0,0 +1,206 @@
+    sleep_sema.signal();
+}
+
+// low priority threaded context
+int main() {
+    // hardware timer, can only execute code in interrupt context
+    Timer timer;
+    timer.attach_ms(irq, 100);
+
+    // attach our sleep callback so we get sleep updates
+    equeue.background(sleep_callback);
+
+    while (true) {
+        // dispatch any pending events
+        equeue.dispatch(0);
+
+        // super duper fancy status update system
+        led = !led;
+
+        // go to sleep until next event
+        sleep_sema.wait(sleep_time);        
+    }
+}
+```
+
+Now you will execute events, toggle your LED and sleep all using just one execution context.
+
+The `background` function handles all of the corner cases for you, such as canceling events and nested events, so as long as you call dispatch after the timeout, everything works.
+
+<span class="notes">**Note:** When the event queue _doesn't_ have a timeout, the callback is passed the value `-1` (which is the same as `osWaitForever`). In this example, passing `-1` to the semaphor wait returns the behavior you want, but sometimes you need to treat this as a special case. Also, the event queue passes `-1` at destruction time.</span>
+
+That while loop resembles an event queue by itself. Although you can use the `background` function to run one event queue in another event queue's context, it's complicated enough that it is wrapped up in `background`'s sister function: `chain`:
+
+``` cpp
+/** Chain an event queue onto another event queue
+ *
+ *  After chaining a queue to a target, calling dispatch on the target
+ *  queue also dispatches events from this queue. The queues use
+ *  its own buffers, and you must handle events independently.
+ *
+ *  A null queue as the target unchains the existing queue.
+ *
+ *  The chain function allows you to compose multiple event queues,
+ *  sharing the context of a dispatch loop while still being managed
+ *  independently.
+ *
+ *  @param target   Queue that dispatches this queue's events as a
+ *                  part of its dispatch loop
+ */
+void chain(EventQueue *target);
+```
+
+With `chain`, you can rewrite [a blinking light example](mbed-os-quick-start.html) to use its own event queue:
+
+``` cpp
+// your own event queue
+EventQueue blinky_equeue(2*EVENTS_EVENT_SIZE);
+
+// status update
+void blink() {
+    led = !led;
+}
+
+// low priority threaded context
+int main() {
+    // hardware timer, can only execute code in interrupt context
+    Timer timer;
+    timer.attach_ms(irq, 100);
+
+    // chain your event queue onto this new event queue for the blinking lights
+    equeue.chain(&blinky_equeue);
+
+    // blink our led every 100 ms
+    blinky_equeue.call_every(100, blink);
+
+    // dispatch our root event queue
+    blinky_equeue.dispatch();
+}
+```
+
+You want to tightly control the memory consumption of each event loop.
+
+Consider an example Sonar class driven by a complicated FSM that I'm too lazy to write:
```

Comment:
Could an engineer please create an example to replace this one?

Reply:
Petition to replace this with a link to the sonar example as per our discussion @geky @AnotherButler

Reply:
@geky @AnotherButler  Thoughts on the sonar example link?

Reply:
Sounds reasonable to me. That or we could just remove the sentence about laziness. The example does have the part that's important to users of the event queue.

I'd remove it but I don't think I can push to this branch.